DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,283 @@
+- Create a pull request that targets the appropriate branch, using the PR description template.
+- Sign off your commits by including a Signed-off-by line in each commit message.
+
+## Repository Relations
+
+This section describes the scope of each repository:
+
+### Main Components
+
+- harvester/harvester
+  - Basic features, including virtual machines, images, upgrades ([package/upgrade](./package/upgrade)), volumes, etc.
+- harvester/harvester-installer
+  - Installation console and ISO building. It also packages the underlying OS (harvester/os2) and rancher/rancherd.
+- harvester/docs
+  - [Harvester Official Documentation](https://docs.harvesterhci.io/)
+- harvester/network-controller-harvester
+  - Manages host network configuration.
+- harvester/terraform-provider-harvester
+  - Terraform provider for Harvester.
+- harvester/node-disk-manager
+  - Provides an automated way to add storage to Longhorn as Harvester's backend storage, including multipath support.
+- harvester/harvester-ui-extension
+  - Harvester Dashboard UI. For more details, see the [official documentation](https://docs.harvesterhci.io/v1.6/rancher/harvester-ui-extension/).
+
+
+### Add-ons
+
+### Built-in
+
+- harvester/pcidevices
+  - PCI, USB, GPU, and vGPU device passthrough.
+- harvester/nvidia-driver-toolkit
+  - Enable vGPU devices and assign them to Harvester virtual machines. It's used with harvester/pcidevices together.
+- harvester/rancher-logging
+  - Collect versatile logs, events, and audits from the Harvester cluster and route them to many kinds of servers based on flows.
+- harvester/rancher-monitoring
+  - Collect Harvester cluster and virtual machine metrics, view them on the embedded dashboard, and send alert(s) to remote servers.
+- harvester/vm-import-controller
+  - Helps migrate VM workloads from external clusters to an existing Harvester cluster. It currently supports VMware and OpenStack.
+
+#### Experimental
+
+- harvester/experimental-addons/harvester-csi-driver-lvm
+  - CSI driver that supports local path provisioning through LVM
+- harvester/experimental-addons/vm-dhcp-controller
+  - A managed DHCP service for virtual machines running on Harvester.
+- harvester/experimental-addons/rancher-vcluster
+  - Runs Rancher Manager as a workload on the underlying Harvester cluster, implemented using vcluster. It's one way to start a Rancher service and use [Rancher Integration](https://docs.harvesterhci.io/v1.5/rancher/rancher-integration) with Harvester.
+
+
+### Rancher-Related
+
+- harvester/docker-machine-driver-harvester
+  - Use Harvester as a cloud provider to provision guest clusters in Rancher. The related UI is located [here](https://github.com/rancher/dashboard/tree/master/pkg/harvester-manager).
+
+### Most Closely Integrated Upstream Repositories
+
+- longhorn/longhorn
+  - Harvester uses Longhorn for virtual machine and node volumes.
+- kubevirt/kubevirt
+  - Harvester uses KubeVirt to provide virtualization.
```

Comment:
Added
- kubevirt/containerized-data-importer
   - Harvester uses Containerized-Data-Importer to provide the third-party storage integration